FP_analysis_of_reviews_from_Amazon/Task_9_3.py

```diff
@@ -0,0 +1,90 @@
+from os.path import dirname, abspath
+from pathlib import Path
+from common_module.work_with_document import get_data_from_json, save_file
+import re
+from collections import Counter
+from operator import itemgetter
+from concurrent.futures import ProcessPoolExecutor
+from common_functions import run_func_parallel
+
+
+def split_text_up(text):
+    """function gets a text and returns list of words from this text"""
+    list_of_words = []
+    for review in text:
+        list_of_words.extend(re.split(r"\W+", review.replace("'", "_").lower()))
```

Comment:
Please, refactor it in FP style

Reply:
I've done it so
```
def split_text_up(text):
    """function gets a text and returns list of words from this text"""
    list_of_words = []
    list(
        map(
            lambda review: list_of_words.extend(
                re.split(r"\W+", review.replace("'", "_").lower())
            ),
            text
        )
    )
    return list_of_words
```
It seems spoiled in any case, because I need get extending result (list of words instead of list of lists)

Reply:
I think we can better. What about this:
```
def text_to_words(text):
    """function gets a text and returns list of words from this text"""
    return re.split(r"\W+", text.replace("'", "_").lower())


def create_words_counter(comments):
    return Counter(
        list(
            sorted(
                [
                    word
                    for word in chain.from_iterable(
                        [text_to_words(comment) for comment in comments]
                    )
                    if word
                ]
            )
        )
    )
```

Reply:
the chain is `from itertools import chain`

Reply:
It is something magical. I hope that I will be able to repeat this in the future.